datalad/support/gitrepo.py

```diff
@@ -157,6 +163,26 @@ def newfunc(self, files, *args, **kwargs):
     return newfunc
 
 
+def _remove_empty_items(list_):
+    """Remove empty entries from list
+
+    This is needed, since some functions of GitPython may convert
+    an empty entry to '.', when used with a list of paths.
+
+    Parameter:
+    ----------
+    files: list of str
+
+    Returns:
+    list of str
+    """
+    if not isinstance(list_, list):
+        lgr.warning(
+            "_remove_empty_items() called with non-list type: %s" % type(files))
+        return files
+    return [file_ for file_ in list_ if file_]
```

Comment:
not nit-picking but rather inspiration - alternative implementation could be
```python
return filter(bool, list_)
```
;-)